chatrooms.lua

```diff
@@ -93,14 +101,37 @@ function ToArrival( tUser, sMessage )
 		Core.SendPmToUser( tUser, sTo, "Sorry! You don't have access to the chatroom.|" )
 		return true
 	else
-		local sChat = sMessage:match "%b$$(.*)|"
-		SendToRoom( tUser, sTo, sChat )
+		local sChat = sMessage:match "%b$$(.*)|" 
+		SaveToFile(sChat, sTo)
+		local sCmd, sData = sMessage:match "%b$$%b<>%s+[-+*/?!#](%w+)%s?(.*)|" 
+		if not sCmd then
+			SendToRoom( tUser, sTo, sChat )
+		elseif sCmd:lower() == "history" then
+			local sReply, sData = "Past %d messages: \n\n\t%s\n\n", tonumber( sData )
+			if (not sData) or sData > 35 or sData < 0 then sData = 15 end
```

Comment:
Use the name `iLimit` in lieu of `sData` as it will either be an integer or `nil` value.